_posts/2021-03-22-keccak-optimizer-bug.md

```diff
@@ -0,0 +1,89 @@
+---
+layout: post
+published: true
+title:  "Solidity Empty Byte Array Copy Bug"
+date:   2021-03-22
+author: Solidity Team
+category: Security Alerts
+---
+
+On March 20, 2021, a bug in Solidity's bytecode optimizer was found. The bug is fixed with [version
```

Comment:
Perhaps I should add that it was found using the fuzzer and link to the fuzzer blog post. @bshastry Anything specific that you would like to add?

Reply:
> @bshastry Anything specific that you would like to add?

Perhaps mention that it was found by differentially testing the old and new code generation engines with fuzzer generated input? Essentially how it was found is that for the original example in https://github.com/ethereum/solidity/issues/11131, the assert in the solidity code passed during an EVM run corresponding to the optimised code generated by the old code generation engine but did not pass for the new code generation engine counterpart.

This differential fuzzing is a recent project that is not covered by the fuzzing blog post though, so perhaps better to refer to it as ongoing work or as a follow-up of the work covered in that blog post.